### PR TITLE
Fixed bug #1, and added unit test for bug scenario.

### PR DESCRIPTION
**Pull request by DustinWoods**

### CODE REVIEW

**Comment by DustinWoods** [comment]
I'm not sure if you will like me second commit, but I utilized Node's built-in url resolver to rewrite your url resolver. A lot of lines removed, but I think Node's url resolving is probably more dependable. You can see documentation here: https://nodejs.org/api/url.html#url_url_resolve_from_to

This pull request is mainly to fix bugs we've found in broken-link-checker. Which is great by the way!

**Comment by stevenvachon** [comment]
Please remove your second commit. I wrote a custom url resolver for a reason ([node#1435](https://github.com/nodejs/node/issues/1435) for example).

File: lib/areSameQuery.js [hunk 1]
```diff
@@ -21,9 +21,9 @@ function areSameQuery(queryObj1, queryObj2)
 		// If strings don't match
 		if (value1 !== value2)
 		{
-			// Only other type allowed is Array
+			// Don't proceed unless both are arrays
 			// "?var1=a&var1=b" would've been parsed to { var1: ["a","b"] }
-			if (Array.isArray(value1) !== Array.isArray(value2)) return false;
+			if (!Array.isArray(value1) || !Array.isArray(value2)) return false;
```
Comment (stevenvachon) [inline lib/areSameQuery.js]: Please replace with:
```js
if (Array.isArray(value1)===false || Array.isArray(value2)===false) return false;
```

**Comment by DustinWoods** [comment]
I didn't realize there was a bug with Node's url resolver. Huh. You've put a lot of detailed work into this! Nice job.

**Comment by stevenvachon** [comment]
Thank you. There is still more work to do and I have been sidetracked by other projects for now. I'll be getting back to this one as I hope to propose this package as a replacement for Node's url.

Thanks for catching this bug! I'll release a new version tonight for broken-link-checker to use.

**Pull request merged**